### PR TITLE
Use TlvStream for encrypted channel backup

### PR DESCRIPTION
**Pull request by t-bast**
In order to support extensions to standard LN messages, we must put our custom fields in a `TlvStream`.
The only field that didn't respect that was the encrypted channel backup.
This change is a required preliminary step to be able to implement https://github.com/lightningnetwork/lightning-rfc/pull/847

We also fix a bug where we didn't send our backup in `shutdown` messages.

### CODE REVIEW

File: src/commonTest/kotlin/fr/acinq/lightning/wire/LightningCodecsTestsCommon.kt [hunk 1]
```diff
@@ -494,56 +495,58 @@ class LightningCodecsTestsCommon : LightningTestSuite() {
...
+            Hex.decode("0027") + channelId.toByteArray() + Hex.decode("00000000075bcd15") + signature.toByteArray() + Hex.decode("fe47010000 07 cccccccccccccc") to ClosingSigned(channelId, 123456789.sat, signature).withChannelData(ByteVector("cccccccccccccc")),
+            Hex.decode("0027") + channelId.toByteArray() + Hex.decode("00000000075bcd15") + signature.toByteArray() + Hex.decode("01 02 0102") + Hex.decode("fe47010000 07 cccccccccccccc") to ClosingSigned(channelId, 123456789.sat, signature, TlvStream(listOf(ClosingSignedTlv.ChannelData(EncryptedChannelData(ByteVector("cccccccccccccc")))), listOf(GenericTlv(1, ByteVector("0102")))))
         )
-        //@formatter:on
+        // @formatter:on
 
         refs.forEach {
-            val decoded = LightningMessage.decode(it.key.first + it.key.second)
+            println(it.key.toByteVector().toHex())
```
Comment (sstone) [inline src/commonTest/kotlin/fr/acinq/lightning/wire/LightningCodecsTestsCommon.kt:545]: nit: remove ?

**Review by sstone — CHANGES_REQUESTED** [review]
LGTM, just a nit

File: src/commonTest/kotlin/fr/acinq/lightning/wire/LightningCodecsTestsCommon.kt (hunk 1 above)
Comment (t-bast) [reply src/commonTest/kotlin/fr/acinq/lightning/wire/LightningCodecsTestsCommon.kt:545]: Good catch!

Comment (t-bast) [reply src/commonTest/kotlin/fr/acinq/lightning/wire/LightningCodecsTestsCommon.kt]: Done in https://github.com/ACINQ/lightning-kmp/pull/260/commits/c3fc0bd3a5451819bb6eca21702080a78890a7fb

**Review by sstone — APPROVED** [review]

**Pull request merged**